sources/api/netdog/src/main.rs

```diff
@@ -280,6 +303,93 @@ fn generate_hostname() -> Result<()> {
     Ok(print_json(hostname)?)
 }
 
+/// Generate configuration for network interfaces.
+fn generate_net_config() -> Result<()> {
+    if !Path::exists(Path::new(NET_CONFIG)) {
+        match fetch_default_interface(KERNEL_CMDLINE)? {
+            Some(interface_str) => {
+                let default_interface = WickedInterface::from_str(&interface_str)
+                    .context(error::InvalidInterfaceCmdlineSnafu)?;
+                write_primary_interface(default_interface.name())?;
+
+                return default_interface
+                    .write_config_file()
+                    .context(error::InterfaceConfigWriteSnafu);
+            }
+            None => {
+                eprintln!("Network interface config not found on kernel command line");
+                return Ok(());
+            }
+        }
+    }
+
+    let net_config =
+        NetConfig::from_path(NET_CONFIG).context(error::NetConfigReadSnafu { path: NET_CONFIG })?;
+    let primary_interface = net_config
+        .primary_interface()
+        .context(error::GetPrimaryInterfaceSnafu)?;
+    write_primary_interface(primary_interface)?;
+
+    for (name, config) in net_config.interfaces {
+        if config.dhcp4.is_none() && config.dhcp6.is_none() {
+            return error::InvalidNetConfigSnafu {
+                interface: name.to_string(),
+                reason: "must configure dhcp4 or dhcp6, or both",
+            }
```

Comment:
If we don't specify either DHCP4 or DHCP6 in the wicked interface XML file, what does wicked do? Bring it up with just local addresses? Leave it down? Error out?